### ISSUE TITLE
[Tama][AOSP10][4.14] Boot on charger hangs the phone or leads to bootloop

### ISSUE THREAD

**Issue by rinigus** [sonyxperiadev/bug_tracker#700]
**Platform:** Tama
**Device:** Apollo
**Kernel version:** 4.14.232-g36679be59282
**Android version:** android-10.0.0_r41
**Software binaries version:** SW_binaries_for_Xperia_Android_10.0.7.1_r1_v12a_tama

**Previously working on**

Has worked on AOSP9, at least in SFOS port.

**Description**

When switching off the phone and powering it on charger, phone will eventually show battery charging animation, as expected. However, depending on USB power source, it will either become unresponsive (charger) or enter a bootloop (if USB is connected to PC).

**How to reproduce**

1. switch off phone
2. insert USB plugged to a charger
3. regular boot sequence will appear - bootloader warning, SONY logo. during SONY animation (or maybe earlier), LED will switch from red to off
4. battery charging animation will be shown
5. screen will switch off

In this state, phone will not react to short power button press, long power button press. If I take the cable out, same will continue. The only way to get phone back is to force reboot by vol up+power combo.

When the same sequence is repeated with USB plugged to PC, phone will get from 1-5 and then reboot with the cycle (3-5) repeated while cable is plugged in. To break bootloop, unplug the cable. Then phone will switch off after showing battery symbol. After that, phone can be booted as usual by pressing power button for longer time.

**Additional context**

Maybe related to https://github.com/sonyxperiadev/bug_tracker/issues/664, but not cosmetic.

Originally filed under SFOS port issues: https://github.com/sailfishos-sony-tama/main/issues/119

**Comment by rinigus** [comment]
On Sailfish, I managed to make it all work. Sailfish is using somewhat different approach: when booting on charger, we switch over to `late-init`. However, while debugging it, I have found that 

- https://github.com/sonyxperiadev/device-sony-common/blob/q-mr1/rootdir/vendor/etc/init/init.usb.rc#L15 looks to be leading to reboot while starting on charger connected to PC
- https://github.com/sonyxperiadev/device-sony-tama/blob/master/rootdir/vendor/etc/init/init.tama.pwr.rc#L15 leads the device hang

On SFOS, we have a script which is bringing selected CPUs to offline and enabling powersave mode. For some reason, that script was also leading to device hang after running it on Apollo.

Hopefully, these comments are helpful.
